dashboard/backend/handler/apihandler.go

```diff
@@ -0,0 +1,170 @@
+package handler
+
+import (
+	"fmt"
+	"net/http"
+
+	restful "github.com/emicklei/go-restful"
+	"github.com/tensorflow/k8s/dashboard/backend/client"
+	"github.com/tensorflow/k8s/pkg/spec"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/pkg/api/v1"
+)
+
+type APIHandler struct {
+	cManager client.ClientManager
+}
+
+//add tensorboard ips, add azure file / gs links
+type TfJobDetail struct {
+	TfJob     *spec.TfJob `json:"tfJob"`
+	TbService *v1.Service `json:"tbService"`
+	Pods      []v1.Pod    `json:"pods"`
+}
+
+type TfJobList struct {
+	tfJobs []spec.TfJob `json:"TfJobs"`
+}
+
+func CreateHTTPAPIHandler(client client.ClientManager) (http.Handler, error) {
+	apiHandler := APIHandler{
+		cManager: client,
+	}
+
+	wsContainer := restful.NewContainer()
+	wsContainer.EnableContentEncoding(true)
+
+	cors := restful.CrossOriginResourceSharing{
+		ExposeHeaders:  []string{"X-My-Header"},
+		AllowedHeaders: []string{"Content-Type", "Accept"},
+		AllowedMethods: []string{"GET", "POST", "DELETE"},
+		CookiesAllowed: false,
+		Container:      wsContainer,
+	}
+	wsContainer.Filter(cors.Filter)
+	wsContainer.Filter(wsContainer.OPTIONSFilter)
+
+	apiV1Ws := new(restful.WebService)
+
+	apiV1Ws.Path("/api").
```

Comment:
How do these API methods differ from the API methods that the K8s APIServer provides for our TfJob CRD?

Reply:
Some of them (such as `DELETE /tfjob/{namespace}/{tfjob}` are basically just forwarded to the APIServer as is.
Others such as `GET /tfjob/{namespace}/{tfjob}` are different and return more data that the APIServer would (such as related pods, service details for TensorBoard etc.)

Reply:
Is there any particular reason why you need to define methods that just forward to the APIServer rather than having callers of those methods just call the APIServer directly?

I think this has the potential of creating confusion down the line. e.g if someone is writing a program to create TfJobs programmatically, should they be using the APIServer or the backend server? 

Should the extra data (e.g. service details for TensorBoard) be included in the TfJob status?

Reply:
This API is called by the dashboard's frontend only.
Since the frontend is just a piece of JS running on the client's end (out of cluster), there is no way to access the APIServer from there securely.   
So the frontend makes a call to the dashboard's backend who is running in-cluster and is authenticated with the APIServer.  
This is similar to what is happening in k8s dashboard: https://github.com/kubernetes/dashboard/blob/master/src/app/backend/handler/apihandler.go#L248

Reply:
The APIServer speaks oauth. So couldn't the JS app running on the client just obtain an OAuth token for the user and then include that in the https request to the APIServer?

That's probably beyond the scope of this PR. However, if that seems like a promising approach then it might be worth opening an issue or TODO to track.